### PR TITLE
79508 - unitofsale and bulkpriceupdate event endpoint authentication

### PR DESCRIPTION
**Pull request by Vishal14583**
- Azure AD Authentication for ERP Facade endpoints
- Functional Tests

### CODE REVIEW

**Review by Sadha1501493 — CHANGES_REQUESTED** [review]
Please do changes as per comment

File: tests/UKHO.ERPFacade.API.FunctionalTests/FunctionalTests/BulkPriceUpdateScenario.cs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+using NUnit.Framework;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using UKHO.ERPFacade.API.FunctionalTests.Helpers;
+using UKHO.ERPFacade.API.FunctionalTests.Model;
+
```
Comment (Sadha1501493) [inline tests/UKHO.ERPFacade.API.FunctionalTests/FunctionalTests/BulkPriceUpdateScenario.cs:13]: Please remove unused statements

File: tests/UKHO.ERPFacade.API.FunctionalTests/Helpers/AzureTableHelper.cs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+using Azure.Data.Tables;
+using Azure;
+using UKHO.ERPFacade.Common.Models.TableEntities;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+
```
Comment (Sadha1501493) [inline tests/UKHO.ERPFacade.API.FunctionalTests/Helpers/AzureTableHelper.cs:11]: Please remove unused statements

File: tests/UKHO.ERPFacade.API.FunctionalTests/Helpers/UnitOfSaleEndpoint.cs [hunk 3]
```diff
@@ -1,24 +1,32 @@
...
 
 namespace UKHO.ERPFacade.API.FunctionalTests.Helpers
 {
     public class UnitOfSaleEndpoint 
     {
 
-        private readonly RestClient client;        
-             
+        private readonly RestClient client;
+        private readonly RestClient client2;
```
Comment (Sadha1501493) [inline tests/UKHO.ERPFacade.API.FunctionalTests/Helpers/UnitOfSaleEndpoint.cs:19]: Same as above

**Review by Sadha1501493 — CHANGES_REQUESTED** [review]

File: tests/UKHO.ERPFacade.API.FunctionalTests/Helpers/BulkPriceUpdateEndpoint.cs [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+using System.Text;
+using System.Threading.Tasks;
+
+namespace UKHO.ERPFacade.API.FunctionalTests.Helpers
+{
+    public class BulkPriceUpdateEndpoint 
+    {
+
+        private readonly RestClient client;
+        private readonly RestClient client2;
```
[the remark this answers is not in the document]
Comment (Sadha1501493) [reply tests/UKHO.ERPFacade.API.FunctionalTests/Helpers/BulkPriceUpdateEndpoint.cs]: Done

File: tests/UKHO.ERPFacade.API.FunctionalTests/Helpers/UnitOfSaleEndpoint.cs [hunk 5]
```diff
@@ -45,6 +53,16 @@ public async Task<RestResponse> PostUoSResponseAsync(string filePath, string tok
...
+            JsonPayloadHelper jsonPayload = JsonConvert.DeserializeObject<JsonPayloadHelper>(requestBody);
+            string traceID = jsonPayload.Data.TraceId;
+
+
+
+            if (response.StatusCode == System.Net.HttpStatusCode.OK)
+            {
+                Assert.That(azureTableHelper.CheckResponseDateTime(traceID), Is.True, "ResponseDateTime Not updated in Azure table");
+            }
+            //Assert.That(azureTableHelper.CheckResponseDateTime(traceID), Is.True, "ResponseDateTime Not updated in Azure table");
```
[the remark this answers is not in the document]
Comment (Sadha1501493) [reply tests/UKHO.ERPFacade.API.FunctionalTests/Helpers/UnitOfSaleEndpoint.cs]: Done

**Review by Sadha1501493 — APPROVED** [review]
reviewed FT related changes

File: src/UKHO.ERPFacade.API/appsettings.json
Comment (Shirin14926) [inline src/UKHO.ERPFacade.API/appsettings.json:1]: Please remove local setup values for this values

File: tests/UKHO.ERPFacade.API.FunctionalTests/Helpers/Config.cs [hunk 6]
```diff
@@ -1,4 +1,6 @@
 ﻿using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
```
Comment (Shirin14926) [inline tests/UKHO.ERPFacade.API.FunctionalTests/Helpers/Config.cs:2]: Remove unused statements and spaces

File: src/UKHO.SAP.MockAPIService/appsettings.json
Comment (Shirin14926) [inline src/UKHO.SAP.MockAPIService/appsettings.json:1]: Similarly for this file as well

File: tests/UKHO.ERPFacade.API.FunctionalTests/Helpers/SAPXmlHelper.cs [hunk 7]
```diff
@@ -19,9 +17,9 @@ public class SAPXmlHelper
 
         static int actionCounter;
 
-        static List<string> AttrNotMatched = new List<string>();
+        static readonly List<string> AttrNotMatched = new List<string>();
         static List<string> ChangeAVCSUoS = new List<string>();
-        static Dictionary<string, List<string>> ChangeENCCell = new Dictionary<string, List<string>>();
+        static readonly Dictionary<string, List<string>> ChangeENCCell = new Dictionary<string, List<string>>();
```
Comment (Shirin14926) [inline tests/UKHO.ERPFacade.API.FunctionalTests/Helpers/SAPXmlHelper.cs:22]: Can be written as 
static readonly Dictionary<string, List<string>> ChangeENCCell = new();

**Review by Shirin14926 — COMMENTED** [review]
Few comments please

File: tests/UKHO.ERPFacade.API.FunctionalTests/Helpers/SAPXmlHelper.cs (hunk 7 above)
Comment (Sadha1501493) [reply tests/UKHO.ERPFacade.API.FunctionalTests/Helpers/SAPXmlHelper.cs]: done

**Comment by RaviK2233** [comment]
@Sadha1501493 Some of the checks have failed. could you please have a look at it. thanks

File: src/UKHO.ERPFacade.API/appsettings.json
Comment (Vishal14583) [reply src/UKHO.ERPFacade.API/appsettings.json:1]: removed

File: src/UKHO.SAP.MockAPIService/appsettings.json
Comment (Vishal14583) [reply src/UKHO.SAP.MockAPIService/appsettings.json:1]: reverted

**Comment by Sadha1501493** [comment]
> @Sadha1501493 Some of the checks have failed. could you please have a look at it. thanks

Fixed, please proceed @RaviK2233

**Review by Shirin14926 — APPROVED** [review]
Thanks @Vishal14583 for fixing comments. LGTM.

**Review by jrippington — APPROVED** [review]
Looks good to me

**Review by jrippington — APPROVED** [review]
Looks good to me

**Review by martyn-fewtrell — APPROVED** [review]
Looks good

**Pull request merged**